src/content/en/2020/pwa.md

```diff
@@ -22,3 +22,257 @@ queries: 14_PWA
+
+A maskable icon ensures that the image fills the entire shape without being letterboxed when adding the progressive web app to the home screen.. [Learn more]([https://web.dev/maskable-icon-audit/](https://web.dev/maskable-icon-audit/)) **0.14%** of sites use this, given that it is a brand new feature, the percentage of adaptation is pretty decent. This being a new feature we were expecting the numbers to be low and for sure shall improve in the coming years.
+
+A service worker enables your web app to be reliable in unpredictable network conditions. [Learn more]([https://web.dev/offline-start-url/](https://web.dev/offline-start-url/)) **0.77%** of sites has an offline start URL.
+
+The service worker is the feature that enables your app to use many Progressive Web App features, such as offline, add to homescreen, and push notifications. [Learn more]([https://web.dev/service-worker/](https://web.dev/service-worker/)) **1.05%** of pages have service workers enabled, bare in mind this for the month of September 2020. Serviceworker helps to achieve offline support, which is the most important feature for a PWA, as flaky networks are the most common issue that the user of web applications face and this is circumvented with Serviceworkers, it is pretty surprising that number was this low for the month under investigation.
+
+A themed splash screen ensures a native like experience when users launch your app from their homescreens. [Learn more]([https://web.dev/splash-screen/](https://web.dev/splash-screen/)) **1.95%** of pages had splash screens.
+
+The browser address bar can be themed to match your site. [Learn more]([https://web.dev/themed-omnibox/](https://web.dev/themed-omnibox/)) **3.98%** of pages had themed omnibox.
+
+Your app should display some content when JavaScript is disabled, even if it's just a warning to the user that JavaScript is required to use the app. [Learn more]([https://web.dev/without-javascript/](https://web.dev/without-javascript/)) **96.23%** pages can work with JavaScript disabled.
+
+
+# Service worker events
+
+In a service worker one can listen for a number of events:  
+
+1.  `install`, which occurs upon service worker installation.
+    
+2.  `activate`, which occurs upon service worker activation.
+    
+3.  `fetch`, which occurs whenever a resource is fetched.
+    
+4.  `push`, which occurs when a push notification arrives.
+    
+5.  `notificationclick`, which occurs when a notification is being clicked.
+    
+6.  `notificationclose`, which occurs when a notification is being closed.
+    
+7.  `message`, which occurs when a message sent via postMessage() arrives.
+    
+8.  `sync`, which occurs when a background sync event occurs.
+    
+
+  
+
+![](https://lh6.googleusercontent.com/saXCVd5ClUVGqA4HF558mPJfTI5YqRoq9rRPQCiPgZOZ5QuCGx2IJuHytFzgN-Zlz1Ppdjd59vF1pk4pZ-Tn8XVg45tCKdsQHVay02-x4Qfio75zmFw7hMT9EvlY3IWXCW_Lue76 "Chart")
```

Comment:
Images on CDN?

Reply:
See the Figures Guide: https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide#embedded-google-sheets-examples

Reply:
Images have been added. Resolving this.